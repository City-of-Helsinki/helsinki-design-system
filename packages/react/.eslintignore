**/*.d.ts
webpack.config.js
.templates
lib/
**/*.stories.tsx
