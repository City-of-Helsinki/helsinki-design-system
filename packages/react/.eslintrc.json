{
  "parser": "@typescript-eslint/parser",
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:react/recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:import/errors",
    "plugin:import/warnings",
    "plugin:import/typescript",
    "plugin:jest/recommended",
    "airbnb",
    "prettier/@typescript-eslint",
    "plugin:prettier/recommended" // this should always be the last item on this array
  ],
  "plugins": ["react", "@typescript-eslint", "prettier", "jest", "import"],
  "parserOptions": {
    "ecmaVersion": 2018,
    "ecmaFeatures": {
      "jsx": true
    }
  },
  "ignorePatterns": ["*.module.css"],
  "rules": {
    "no-use-before-define": "off",
    "@typescript-eslint/no-use-before-define": ["error"],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": ["**/*.test.ts", "**/*.test.tsx", "**/test*.ts*", "**/*.stories.tsx"]
      }
    ],
    "import/order": [
      "error",
      {
        "groups": ["builtin", "external", "internal", ["parent", "sibling", "index"]],
        "newlines-between": "always"
      }
    ],
    "react/jsx-filename-extension": "off",
    "react/jsx-one-expression-per-line": "off",
    "react/jsx-props-no-spreading": "off",
    "react/jsx-curly-newline": "off",
    "@typescript-eslint/explicit-function-return-type": "off",
    "react/prop-types": "off",
    "import/extensions": "off",
    "import/prefer-default-export": "off",
    "jsx-a11y/label-has-associated-control": [
      2,
      {
        "components": ["Label"],
        "required": {
          "some": ["nesting", "id"]
        },
        "allowChildren": false
      }
    ],
    "@typescript-eslint/explicit-member-accessibility": "off",
    "react/state-in-constructor": [2, "never"],
    "no-unused-expressions": [2, {
      "allowTernary": true
    }],
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": ["error"]
  },
  "settings": {
    "react": {
      "version": "detect"
    },
    "import/resolver": {
      "node": {
        "extensions": [".js", ".jsx", ".ts", ".tsx"],
        "paths": ["src"]
      }
    }
  },
  "env": {
    "browser": true,
    "jest": true
  },
  "globals": {
    "JSX": true
  }
}
