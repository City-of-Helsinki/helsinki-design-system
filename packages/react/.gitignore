node_modules/
storybook-static/
lib/
_bundles/
*.css.d.ts
.loki/current
.loki/difference

*storybook.log
