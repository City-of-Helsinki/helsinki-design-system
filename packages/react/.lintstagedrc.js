module.exports = {
  'src/**/*.{ts,tsx}': 'eslint',
}
