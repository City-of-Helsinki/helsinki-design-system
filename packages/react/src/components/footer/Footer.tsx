import React from 'react';

import '../../styles/base.module.css';
import styles from './Footer.module.scss';
import { Koros, KorosType, getShapeHeight } from '../koros';
import classNames from '../../utils/classNames';
import { FooterNavigation } from './components/footerNavigation/FooterNavigation';
import { FooterNavigationGroup } from './components/footerNavigationGroup/FooterNavigationGroup';
import { FooterGroupHeading } from './components/footerGroupHeading/FooterGroupHeading';
import { FooterLink } from './components/footerLink/FooterLink';
import { FooterUtilities } from './components/footerUtilities/FooterUtilities';
import { FooterUtilityGroup } from './components/footerUtilityGroup/FooterUtilityGroup';
import { FooterBase } from './components/footerBase/FooterBase';
import { FooterCustom } from './components/footerCustom/FooterCustom';
import { FooterTheme } from './Footer.interface';
import { useTheme } from '../../hooks/useTheme';
import { AllElementPropsWithoutRef } from '../../utils/elementTypings';

export type FooterProps = React.PropsWithChildren<{
  /**
   * Additional class names to apply to the footer.
   */
  className?: string;
  /**
   * Props that will be passed to the native `<footer>` element.
   */
  footerProps?: AllElementPropsWithoutRef<'footer'>;
  /**
   * Koros type to use in the footer.
   * @default 'basic'
   */
  korosType?: KorosType;
  /**
   * Defines the footer theme.
   * @default 'light'
   */
  theme?: FooterTheme;
  /**
   * The title of the service shown on top of the the footer.
   */
  title?: React.ReactNode;
}>;

export const Footer = ({
  children,
  className,
  footerProps,
  korosType = 'basic',
  theme = 'light',
  title,
}: FooterProps) => {
  // custom theme class that is applied to the root element
  const customThemeClass = useTheme<FooterTheme>(styles.footer, theme);
  const korosHeight = getShapeHeight({ type: korosType });

  return (
    <footer
      {...footerProps}
      className={classNames(
        styles.footer,
        typeof theme === 'string' && styles[`theme-${theme}`],
        customThemeClass,
        footerProps && footerProps.className,
        className,
      )}
    >
      <Koros className={classNames(styles.koros)} type={korosType} style={{ height: `${korosHeight}px` }} />
      <div className={styles.footerContent}>
        <div className={styles.footerSections}>
          {title && <h2 className={styles.title}>{title}</h2>}
          {children}
        </div>
      </div>
    </footer>
  );
};

Footer.Navigation = FooterNavigation;
Footer.NavigationGroup = FooterNavigationGroup;
Footer.GroupHeading = FooterGroupHeading;
Footer.Link = FooterLink;
Footer.Utilities = FooterUtilities;
Footer.UtilityGroup = FooterUtilityGroup;
Footer.Base = FooterBase;
Footer.Custom = FooterCustom;
