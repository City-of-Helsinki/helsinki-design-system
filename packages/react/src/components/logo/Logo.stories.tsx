import React from 'react';

import { Logo, logoFi, logoSv, logoRu } from './Logo';

export default {
  component: Logo,
  title: 'Components/Logo',
  parameters: {
    controls: { hideNoControlsWarning: true },
  },
};

export const Playground = (args) => <Logo alt="logo" size={args.size} src={args.src} />;

Playground.args = {
  size: 'full',
  src: logoFi,
};

Playground.argTypes = {
  size: {
    options: ['full', 'small', 'medium', 'large'],
    control: { type: 'radio' },
  },
  src: {
    options: [logoFi, logoSv, logoRu],
    control: { type: 'radio' },
  },
};
