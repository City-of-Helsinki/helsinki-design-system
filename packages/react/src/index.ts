export * from './icons';
export * from './components';
export * from './ssr';
