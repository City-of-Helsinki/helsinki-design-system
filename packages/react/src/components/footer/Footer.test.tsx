import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';

import { Footer } from './Footer';
import { Logo } from '../logo';

describe('<Footer /> spec', () => {
  it('renders the component', () => {
    const { asFragment } = render(
      <Footer title="Bar">
        <Footer.Base
          copyrightHolder="Copyright"
          copyrightText="All rights reserved"
          backToTopLabel="YlÃ¶s"
          logo={<Logo alt="logo" size="medium" aria-hidden="true" title="Helsingin kaupunki" src="dummyPath" />}
        />
      </Footer>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not have basic accessibility issues', async () => {
    const { container } = render(<Footer title="Bar" />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
