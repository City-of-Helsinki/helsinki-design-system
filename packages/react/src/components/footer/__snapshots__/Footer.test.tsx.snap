// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> spec renders the component 1`] = `
<DocumentFragment>
  <footer
    class="footer koros-basic theme-light"
  >
    <div
      class="koros basic koros basic"
    >
      <svg
        aria-hidden="true"
        height="85"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <pattern
            height="85"
            id="koros_basic-1"
            patternUnits="userSpaceOnUse"
            width="106"
            x="0"
            y="0"
          >
            <path
              d="M0,800h20V0c-4.9,0-5,2.6-9.9,2.6S5,0,0,0V800z"
              transform="scale(5.3)"
            />
          </pattern>
        </defs>
        <rect
          fill="url(#koros_basic-1)"
          height="85"
          width="100%"
        />
      </svg>
    </div>
    <div
      class="footerContent"
    >
      <div
        class="footerSections"
      >
        <h2
          class="title"
        >
          Bar
        </h2>
        <div
          class="base"
        >
          <hr
            aria-hidden="true"
            class="divider"
          />
          <div
            class="logoWrapper"
          >
            <a
              class="link linkM item"
              tabindex="0"
            >
              <img
                alt="logo"
                aria-hidden="true"
                class="logo medium"
                src="dummyPath"
                title="Helsingin kaupunki"
              />
            </a>
          </div>
          <div
            class="copyright"
          >
            <span
              class="copyrightHolder"
            >
              © Copyright 2023
            </span>
            <span
              class="copyrightDot"
            >
              •
            </span>
            <span
              class="copyrightText"
            >
              All rights reserved
            </span>
          </div>
          <div
            class="baseActions"
          >
            <button
              class="backToTopButton"
              role="link"
              type="button"
            >
              Ylös
              <svg
                aria-hidden="true"
                aria-label="arrow-up"
                class="icon s"
                role="img"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M12 3.5L18.5 10L17 11.5L13 7.5V20H11V7.5L7 11.5L5.5 10L12 3.5Z"
                  fill="currentColor"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </footer>
</DocumentFragment>
`;
